jupyter_bifrost/bifrost.py

```diff
@@ -130,10 +130,33 @@ def update_dataset(self, changes):
         )
 
     def get_data(self, df: pd.DataFrame, sampleLimit: int = None):
+        nan_columns = df.columns[df.isna().all()].tolist()
+        if len(nan_columns):
+            raise SystemError(f"{nan_columns} in this dataset have only Na's")
+
         if sampleLimit and sampleLimit < len(df):
-            df = df.sample(n=sampleLimit)
+            df = self.get_non_na(df, sampleLimit)
         return json.loads(df.to_json(orient="records"))
 
+    def get_non_na(self, df: pd.DataFrame, sampleLimit: int = None):
+        indices = self.get_each_valid_data(df)
+        if sampleLimit:
+            return pd.concat(
+                [
+                    df.loc[indices, :],
+                    df.drop(indices).sample(n=(sampleLimit - len(indices))),
```

Comment:
Could the `sample` call possibly grab NaN values? Not sure if this is an issue. What do you think?

Reply:
It could grab NaN values. But, since we know that each column has at least one non-NaN, it shouldn't be a problem.